src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -747,6 +853,38 @@ private TcpConnectionInfo GetConnectionInfo(ReplicationNode node, bool external)
             return null;
         }
 
+        public X509Certificate2 GetCertificateForReplication(ReplicationNode node, out TcpConnectionHeaderMessage.AuthorizationInfo authorizationInfo)
+        {
+            authorizationInfo = null;
+            var exNode = node as ExternalReplication;
+            if (exNode == null)
+                return _server.Server.Certificate.Certificate;
+
+            if (exNode.PullReplicationAsEdgeSettings?.CertificateThumbprint != null)
```

Comment:
No, we don't want this.
Instead, we want to have a `Certificate` property on the `ExternalReplication` directly that would hold the BASE64 certificate (including private key)

Reply:
Actually, better would be to hold the actual certificate elsewhere (because the certificate private key is... private, but we need:
* To upload the actual pull certificate when creating the pull replication
* To prevent it from being downloaded